### ISSUE TITLE
CKV_OPENAPI_9 should only check for OAuth2 security scheme in version 2.0

### ISSUE THREAD

**Issue by afaiq** [bridgecrewio/checkov#5255]
**Description**
CKV_OPENAPI_9 generates false positives because it uses every type of security scheme in securityDefinitions to validate if scope exists. As stated in swagger official [documentation](https://swagger.io/docs/specification/2-0/authentication/), scopes only used when defining OAuth2 security scheme. However, this block of [code](https://github.com/bridgecrewio/checkov/blob/e2aa231efeed33e1a9dff389f70a5dd652f0f5fc/checkov/openapi/checks/resource/v2/OperationObjectSecurityScopeUndefined.py#L54C39-L54C39) that checks if a certain auth_definition have scope, does not check the type of auth scheme it is first.

I propose adding auth type/scheme validation before checking if the authentication have scope in it.
```python
definition_type = auth_definition.get('type', {})
if definition_type !="oauth2":
     continue
```

**Example**
Given an openapi file that has endpoint with security scheme of API key:
```yaml
/endpoint:
    get:
      tags:
        - some-endpoint
      security:
        - APIKey: []
   
securityDefinitions:
  APIKey:
    type: apiKey
    in: header
    name: X-api-key
    description: |
      Unique api key
```
CKV_OPENAPI_9 should not flag this as an issue

**Version:**
Checkov Version 2.3.3 (check file has no changes when compared to 2.3.301)

**Comment by gruebel** [comment]
hey @afaiq thanks for reaching out.

Sounds like a good change, are you interested in contributing it?

**Comment by afaiq** [comment]
Sure @gruebel, I'm happy to contribute. Is there any technical guideline that I should follow other than [this](https://github.com/bridgecrewio/checkov/blob/main/CONTRIBUTING.md)?

**Comment by gruebel** [comment]
yeah, that's it. You already know, where to change the logic and you can find [here](https://github.com/bridgecrewio/checkov/tree/main/tests/openapi/checks/resource/v2/example_OperationObjectSecurityScopeUndefined) the folder with the tests.
